forage-android/src/main/java/com/joinforage/forage/android/network/model/Balance.kt

```diff
@@ -0,0 +1,40 @@
+package com.joinforage.forage.android.network.model
+
+import org.json.JSONObject
+
+/**
+ * A model that represents the available balance on a customer’s card.
+ * @see [EbtBalance]
+ */
+interface Balance
```

Comment:
I'm totally aligned on the value of putting out the `interface Card` right now along side the `class EbtCard` because I agree that every card will have a `last4`.

I do wonder if the same pattern is necessary for balances if, out of the gate, we can't identify any commonalities that the `EbtBalance` would share with another balance. The need to add an empty interface to our public API feels a bit preemptive and it's something we can't take back. Thoughts on not rolling out the `interface Balance` right now and only rolling it out later if we ever identify something common across all balances?

Reply:
@devinmorgan 

> only rolling it out later if we ever identify something common across all balances

- We want clients to design/build around generic-named types/interfaces.
- If we don't support this interface from the beginning, we have to:
  - now: a) make the return type of `toBalance() -> EbtBalance // non-generic name :(`
  - in the future: b) introduce the `interface Balance` to support `toBalance() -> Balance // breaking change :(` 

> we can't identify any commonalities that the

I see a world where `updatedAt / updated_at` is a common field between different kinds of balances (like it is in the [JS SDK](https://github.com/teamforage/fe-monorepo/blob/79e6dade8cd21c83f43ce14dd7e96b4ca0371281/apps/sdk/src/common/interfaces/Forage/types.ts#L344)). I didn't expose it in this PR, because the previous SDK version did not expose `updated_at`, and we can introduce it when it is requested / future PR.

Reply:
> * If we don't support this interface from the beginning, we have to:
>   
>   * now: a) make the return type of `toBalance() -> EbtBalance // non-generic name :(`
>   * in the future: b) introduce the `interface Balance` to support `toBalance() -> Balance // breaking change :(`

Ah OK good point about needing to change the method signature of `toBalance()` if we don't release it now. I'm convinced!

One follow up ask is if can we update `FlowBalanceViewModel.kt` to use reflection like `if (balance is EbtBalance) { balance as EbtBalance }` instead of just straightaway casting `balance as EbtBalance`

This way the Sample App will also document the conventions that we're advocating merchants follow so we don't forget internal. That OK with you?

Reply:
Yes! Thank you for the thoughtful discourse and attention to detail. Nice catch!